### DISCUSSION TITLE
The Link prop 'to' not accepting the string type in 1.20.1 release.

### DISCUSSION THREAD

**Discussion by mykhailoalekseiev** [TanStack/router#1332] [Q&A] [answered]
I decided to update my packages with @tanstack/react-router, and my linter displayed to me that the string type is not acceptable for the 'to' prop for Link components and navigate calls. I have the auth router logic from [documentation](https://tanstack.com/router/latest/docs/framework/react/examples/authenticated-routes-context). Right now, I see the same error in the code sandbox example.

How can I define the type of redirect link to satisfy Typescript?

**Comment by schiller-manuel** [comment]
you can make the string strictly typed like this:

https://stackblitz.com/edit/tanstack-router-8nkdck?file=src%2Froutes%2Flogin.tsx%3AL14

**Reply by mykhailoalekseiev** [reply]
Yes, it can help. 
But I asked mostly about solution without hardcoding all routes like that and with keeping type safety.

**Reply by schiller-manuel** [reply, marked as answer]
`to` as string should now work again as of https://github.com/TanStack/router/releases/tag/v1.20.4
